cli/util/cancel.go

```diff
@@ -0,0 +1,72 @@
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/nspcc-dev/neo-go/cli/options"
+	"github.com/nspcc-dev/neo-go/cli/smartcontract"
+	"github.com/nspcc-dev/neo-go/pkg/core/transaction"
+	"github.com/nspcc-dev/neo-go/pkg/neorpc/result"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/actor"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm/opcode"
+	"github.com/urfave/cli"
+)
+
+func cancelTx(ctx *cli.Context) error {
+	args := ctx.Args()
+	if len(args) == 0 {
+		return cli.NewExitError("transaction hash is missing", 1)
+	} else if len(args) > 1 {
+		return cli.NewExitError("only one transaction hash is accepted", 1)
+	}
+
+	txHash, err := util.Uint256DecodeStringLE(strings.TrimPrefix(args[0], "0x"))
+	if err != nil {
+		return cli.NewExitError(fmt.Sprintf("invalid tx hash: %s", args[0]), 1)
+	}
+
+	gctx, cancel := options.GetTimeoutContext(ctx)
+	defer cancel()
+
+	c, err := options.GetRPCClient(gctx, ctx)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("failed to create RPC client: %w", err), 1)
+	}
+
+	mainTx, _ := c.GetRawTransactionVerbose(txHash)
+	if !mainTx.Blockhash.Equals(util.Uint256{}) {
```

Comment:
Check firstly if `mainTx` is not nil, because it can be nil in case if it's missing in the memory pool of your RPC node.

Reply:
It's OK not to have `mainTx` in the provided RPC node. But if it's there, then we need to check its `Blockhash`. I.e.:
```
if mainTx != nil && !mainTx.Blockhash.Equals(util.Uint256{}) {
    return cli.NewExitError(fmt.Errorf("transaction %s is already accepted at block %s", txHash, mainTx.Blockhash.StringLE()), 1)
}
```
And after that we need to adapt the rest of the code for case when `mainTx` is nil.